custom/abdm/commcare_extensions.py

```diff
@@ -0,0 +1,17 @@
+from corehq.toggles import RESTORE_ADD_ABDM_TOKEN
+from casexml.apps.phone.xml import get_custom_user_data_for_restore
+from custom.abdm.models import ABDMUser
+
+
+def get_abdm_api_token(username, domain):
+    user, _ = ABDMUser.objects.get_or_create(username=username, domain=domain)
+    if not user.access_token:
+        user.generate_token()
+    return user.access_token
+
+
+@get_custom_user_data_for_restore.extend()
+def get_abdm_user_data(restore_user):
+    if RESTORE_ADD_ABDM_TOKEN.enabled(restore_user.domain):
+        return {"abdm_api_token": get_abdm_api_token(restore_user.username, restore_user.domain)}
```

Comment:
@mkangia In relation to other comment where token is extracted from a list, we are returning a dict from here but on the receiving end, we get it wrapped in a list.

Reply:
Got it :+1: